include/deal.II/lac/sparse_direct.h

```diff
@@ -26,6 +26,17 @@
 #include <deal.II/lac/sparse_matrix_ez.h>
 #include <deal.II/lac/block_sparse_matrix.h>
 
+// include UMFPACK file.
+#ifdef DEAL_II_WITH_UMFPACK
+#  include <umfpack.h>
+#endif
+
+// The following block is a fix for
+// either not having UMFPACK
+// or using CLang with UMFPACK where apparently the symbol is not defined (maybe due to the symbol being in extern "C" block)?
```

Comment:
I do not understand.

This is a simple preprocessor definition guard and should be self evident without a comment.